#! /bin/sh -e
## 11_cobalt-tulip.dpatch by Peter Horton <pdh@colonel-panic.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Cobalt tulip ethernet for 2.4.

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

diff -urpN linux-2.4.27-r5k-cobalt/arch/mips/cobalt/pci.c linux-2.4.27-r5k-cobalt-2700/arch/mips/cobalt/pci.c
--- linux-2.4.27-r5k-cobalt/arch/mips/cobalt/pci.c	2004-11-07 10:53:19.000000000 +0000
+++ linux-2.4.27-r5k-cobalt-2700/arch/mips/cobalt/pci.c	2004-11-28 10:14:46.000000000 +0000
@@ -132,6 +132,8 @@ static void qube_raq_tulip_fixup(struct 
 		pci_write_config_byte(dev, PCI_INTERRUPT_LINE,
 				      COBALT_ETH0_IRQ);
 		dev->irq = COBALT_ETH0_IRQ;
+		if(dev->device == PCI_DEVICE_ID_DEC_TULIP_PLUS)
+			dev->irq = COBALT_ETH1_IRQ;
 
 		dev->resource[0].start = 0x100000;
 		dev->resource[0].end = 0x10007f;
@@ -263,6 +265,7 @@ qube_pcibios_fixup(struct pci_dev *dev)
 struct pci_fixup pcibios_fixups[] = {
 	{ PCI_FIXUP_HEADER, PCI_VENDOR_ID_VIA, PCI_DEVICE_ID_VIA_82C586_1, qube_raq_via_bmIDE_fixup },
 	{ PCI_FIXUP_HEADER, PCI_VENDOR_ID_DEC, PCI_DEVICE_ID_DEC_21142, qube_raq_tulip_fixup },
+	{ PCI_FIXUP_HEADER, PCI_VENDOR_ID_DEC, PCI_DEVICE_ID_DEC_TULIP_PLUS, qube_raq_tulip_fixup },
 	{ PCI_FIXUP_HEADER, PCI_VENDOR_ID_GALILEO, PCI_ANY_ID, qube_raq_galileo_fixup },
 	{ PCI_FIXUP_HEADER, PCI_VENDOR_ID_NCR, PCI_DEVICE_ID_NCR_53C860, qube_raq_scsi_fixup },
 	{ PCI_FIXUP_HEADER, PCI_ANY_ID, PCI_ANY_ID, qube_pcibios_fixup }
